scipy/optimize/_optimize.py

```diff
@@ -2951,11 +2971,11 @@ def bracket(func, xa=0.0, xb=1.0, args=(), grow_limit=110.0, maxiter=1000):
                 xb = w
                 fa = fb
                 fb = fw
-                return xa, xb, xc, fa, fb, fc, funcalls
+                break
```

Comment:
Break rather than returning immediately so we can make sure the bracket is valid.